pkg/proxy/iptables/proxier.go

```diff
@@ -772,12 +788,19 @@ func (proxier *Proxier) syncProxyRules() {
 	// Install the kubernetes-specific postrouting rules. We use a whole chain for
 	// this so that it is easier to flush and change, for example if the mark
 	// value should ever change.
-	writeLine(proxier.natRules, []string{
+	masqRule := []string{
 		"-A", string(kubePostroutingChain),
 		"-m", "comment", "--comment", `"kubernetes service traffic requiring SNAT"`,
 		"-m", "mark", "--mark", proxier.masqueradeMark,
 		"-j", "MASQUERADE",
-	}...)
+	}
+	if proxier.masqueradeRandomFully {
```

Comment:
Please add an e2e to be sure this is parsed correctly when this feature is enabled

Reply:
The existing code will log an error if there is any problem with any rule.

Are there already e2e test(s) that exercise kube-proxy and check that this error is not logged?  If so, then this is already covered.  If not, is it fair to ask for this PR to introduce that?

Reply:
That may be the case, could you reference the kube-proxy test that leverages this generated code for the test cases

I'm mainly trying to find unit test that just validates the expected lines added by this flag or an e2e that enables the flag and ensures the baseline iptables functionality is correct. I can imagine that having the e2e for the scale tests would be useful to see if we can replicate the issue without this flag enabled.

Reply:
The code isn't that confusing, though maybe it would be nice to just have `pkg/util/iptables/` just check the version once and export a bunch of variables like `SupportsCheck` and `SupportsRandomFully` for everyone else to use.

Reply:
On the original point, of testing:

1. I added a unit test that the generated `-j MASQUERADE` includes `--random-fully` _if_ the iptables version supports it.

2. The existing e2e tests surely test that all the generated rules parse, for the version of iptables where the tests run.  If any rule does not parse then none are installed, which causes massive damage --- for example, no client will be able to reach the `kubernetes` Service.